interface/modules/custom_modules/oe-module-weno/scripts/weno_log_sync.php

```diff
@@ -13,55 +13,109 @@
-        die;
+        handleDownloadError("Background Initiated Pharmacy download attempt failed. Internet problem!");
+    }
+
+    if ($isKey === false) {
+        requireGlobals();
     }
-    error_log('Background Initiated Encryption Verify returned: ' . text($isKey == '1' ? 'Verified key is valid.' : 'Invalid Key'));
 
-    $cryptoGen = new CryptoGen();
-    $localPharmacyJson = new WenoPharmaciesJson($cryptoGen);
-    // Check if the background service is active. Intervals are set to once a day
+    $localPharmacyJson = new WenoPharmaciesJson(new CryptoGen());
     $value = $localPharmacyJson->checkBackgroundService();
+
     if ($value == 'active' || $value == 'live') {
-        error_log('Background Initiated Pharmacy Download Started.');
-
-        $status = $localPharmacyJson->storePharmacyDataJson();
-
-        EventAuditLogger::instance()->newEvent(
-            "pharmacy_background",
-            $_SESSION['authUser'],
-            $_SESSION['authProvider'],
-            1,
-            "Background Initiated Pharmacy Download Completed with Status:"  . text($status)
-        );
-        error_log('Background Initiated Weno Pharmacies download completed with status:' . text($status));
-        die;
+        performPharmacyDownload($localPharmacyJson);
     }
 }
 
 /**
+ * Perform the pharmacy data download.
+ *
+ * @param WenoPharmaciesJson $localPharmacyJson
+ */
+function performPharmacyDownload(WenoPharmaciesJson $localPharmacyJson)
+{
+    error_log('Background Initiated Pharmacy Download Started.');
+
+    $status = $localPharmacyJson->storePharmacyDataJson();
+
+    EventAuditLogger::instance()->newEvent(
+        "pharmacy_background",
+        $_SESSION['authUser'],
+        $_SESSION['authProvider'],
+        1,
+        "Background Initiated Pharmacy Download Completed with Status:" . text($status)
+    );
+
+    $wenoLog = new WenoLogService();
+    $wenoLog->insertWenoLog("pharmacy", text($status));
+    error_log('Background Initiated Weno Pharmacies download completed with status:' . text($status));
+}
+
+/**
+ * Download Weno Prescription log.
+ *
  * @throws Exception
  */
 function downloadWenoPrescriptionLog(): void
 {
+    $wenoValidate = new WenoValidate();
+    $isKey = $wenoValidate->validateAdminCredentials(true);
+
+    if ((int)$isKey >= 998) {
+        handleDownloadError("Prescription download attempt failed. Internet problem!");
+    }
+
+    if ($isKey === false) {
+        requireGlobals();
+    }
+
     $logSync = new LogProperties();
     if (!$logSync->logSync()) {
         error_log("Background services failed for prescription log.");
     }
 }
+
+/**
+ * Handle download errors.
+ *
+ * @param string $errorMessage
+ */
+function handleDownloadError(string $errorMessage)
+{
+    EventAuditLogger::instance()->newEvent(
+        "pharmacy_background",
+        $_SESSION['authUser'],
+        $_SESSION['authProvider'],
+        1,
+        text($errorMessage)
+    );
+
+    error_log($errorMessage);
```

Comment:
Would html escape via attr() or use the wrapper errorLogEscape() on this. I agree the road is windy to have a vulnerability from this,  however, there is no downside to doing this and we have been doing this for awhile as standard process. At the least, it will avoid dealing with reported vulnerabilities regarding this issue and will make the folks that deal with security reports lives easier.